For Trash context menu in 'dolphin desktop:/'.
This is not related to virtual Trash in kio_desktop, just an improvement.
diff -Nur dolphin-15.12.0/src/dolphincontextmenu.cpp dolphin-15.12.0n/src/dolphincontextmenu.cpp
--- dolphin-15.12.0/src/dolphincontextmenu.cpp	2015-12-08 23:49:30.000000000 +0800
+++ dolphin-15.12.0n/src/dolphincontextmenu.cpp	2015-12-23 19:09:44.874010257 +0800
@@ -97,7 +97,12 @@
         m_context |= TrashContext;
     }
 
-    if (!m_fileInfo.isNull() && !m_selectedItems.isEmpty()) {
+    //virtual Trash in kio desktop
+    if (!m_fileInfo.isNull() && m_fileInfo.url() == QUrl("trash:/")) {
+        m_context |= TrashContext; 
+    }
+
+    if (!m_fileInfo.isNull() && !m_selectedItems.isEmpty() && (m_fileInfo.url() != QUrl("trash:/"))) {
         m_context |= ItemContext;
         // TODO: handle other use cases like devices + desktop files
     }
